### PR TITLE
Support for multi-world providers

### PR DESCRIPTION
**Pull request by JustTalDevelops**

### CODE REVIEW

File: server/player/playerdb/dimension.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import "github.com/df-mc/dragonfly/server/world"
+
+const (
+	overworld = uint8(iota)
+	nether
+	end
+)
+
+func dimensionToID(dimension world.Dimension) uint8 {
```
Comment (Sandertv) [inline server/player/playerdb/dimension.go:11]: Why not just dimension.EncodeDimension?

Comment (JustTalDevelops) [reply server/player/playerdb/dimension.go]: Oops, removed this.

File: server/server.go [hunk 2]
```diff
@@ -342,31 +346,37 @@ func (srv *Server) finaliseConn(ctx context.Context, conn session.Conn, l Listen
 func (srv *Server) defaultGameData() minecraft.GameData {
 	return minecraft.GameData{
 		// We set these IDs to 1, because that's how the session will treat them.
-		EntityUniqueID:               1,
-		EntityRuntimeID:              1,
-		Difficulty:                   2,
```
Comment (Sandertv) [inline server/server.go:347]: Any reason to remove the difficulty normal?

Comment (JustTalDevelops) [reply server/server.go:347]: No particular reason, readded.

**Review by Sandertv — APPROVED** [review]
I think this looks good now, thanks!

**Pull request merged**